Make module_name_rx extendable

rurban opened the pull request:
One might not accept the parser-based limitations of valid module names, e.g.
by creating packages in XS. E.g I use the int? type as type alias for (int | undef),
other might want to use unicode names.

This way you only need to asjust one variable: $Module::Runtime::module_name_rx

karenetheridge commented:
Shouldn't the alteration of `$Module::Runtime::module_name_rx` be localized to the scope in which it's needed? otherwise, "interesting" things could happen elsewhere in the program.

I do support the desire to change the regex that is used, but I'm not sure if this is the way to do it.

rurban commented:
Yes, you could localize your change to this package variable. This module
however just optionally picks such a change to the variable in the other
package.
On Jun 9, 2015 10:53, "Karen Etheridge" notifications@github.com wrote:

> Shouldn't the alteration of $Module::Runtime::module_name_rx be localized
> to the scope in which it's needed? otherwise, "interesting" things could
> happen elsewhere in the program.
> 
> I do support the desire to change the regex that is used, but I'm not sure
> if this is the way to do it.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/doy/package-stash-xs/pull/6#issuecomment-110430304.

ribasushi commented:
Given this patch has a very niche application and tangibly complicates the code (adding a hardcoded copy of a part of `Module::Runtime`, with an implicit logic fork depending if `M::R` has been loaded), I recommend holding off until https://github.com/doy/package-stash/issues/15 is resolved, at which point the only relevant part of this PR would be the (expected to pass) addition to `t/extension.t`

karenetheridge committed:
get build working again

Dist::Zilla::Plugin::MakeMaker 5.022 shifted to creating its file in the
FileGatherer phase, which is earlier than when PruneFiles runs, which
removed the small in-repo Makefile.PL so it can be replaced with the
generated version. Now we must prevent it from being gathered in the
first place.

karenetheridge committed:
add Changes entry for PR#2

karenetheridge committed:
Revert "stop using Test::Requires here"

This reverts commit 33a055a139517143c2975d2eae8964b5a5e9bd16.

It's okay to have this as a requirement when RELEASE_TESTING=1, since we
normally release from a very recent perl and the variable is never set
otherwise.

karenetheridge committed:
convert packaging to my plugin bundle

in this commit, most plugins use the same config as before, so diffs are
kept to a minimum

karenetheridge committed:
document repository shift

karenetheridge committed:
authority rests with STEVAN now

karenetheridge committed:
add Stevan to authors list

karenetheridge committed:
add author test, testing against the proper version

karenetheridge committed:
make Variable::Magic a suggested prereq, and add to develop-requires

karenetheridge committed:
revert to standard weaver configs

...which necessitates removing trailing code so all the pod doesn't get
relocated.

karenetheridge committed:
this code can be simplified since $VERSION is always set now

karenetheridge committed:
remove pod section with redundant and outdated links

karenetheridge committed:
revert to stricter configs for this test

karenetheridge committed:
add a passing spelling test

karenetheridge committed:
remove unnecessary shebangs

karenetheridge committed:
switch from Test::Requires to Test::Needs

karenetheridge committed:
modernize travis config

karenetheridge committed:
generate ppport.h via Devel::PPPort

..and the .h file will be updated in the repository after each release

karenetheridge committed:
check for conflicts during testing, and check in Moose too

karenetheridge committed:
more args for local dev builds

karenetheridge committed:
increment $VERSION after 0.29 release

karenetheridge committed:
Package-Stash-XS-0.29

      - quiet compiler warning (Jacques Germishuys, PR #2)
      - canonical repository moved to
        https://github.com/moose/Package-Stash=XS

karenetheridge committed:
capitalize abstract

karenetheridge commented:
This distribution now lives at https://github.com/moose/Package-Stash-XS -- let's finish this up there.
